templates/components/products/list-item.html

```diff
@@ -4,6 +4,22 @@
     <article class="listItem">
 {{/if}}
     <figure class="listItem-figure">
+        {{#or price.non_sale_price_with_tax price.non_sale_price_without_tax}}
+        {{#if theme_settings.product_sale_badges '===' 'sash'}}
+            <div class="sale-flag-sash">
+                <span class="sale-text">On Sale!</span>
```

Comment:
Is there a reason we need these span tags ? Also I this is going to be surfaced in the theme the I think we need to move the text into `schema.json` and `config.json` so we merchants can modify them.

thoughts @bookernath @bc-as ??

Reply:
span tags used for text styles and aligning. We have already used this HTML construction for sale badges in products grid. Of course we can add possibility to our merchants to edit sale badge text. But I suppose that we need to add an input for it control panel firstly. To allow merchants set up it. In this pull request I just wanted to fix the bug with lack of sale badge in products list

Reply:
@junedkazi we historically haven't exposed the text on those badges to be configurable, but it is a good idea.  I'll open another ticket to address that feature.

Reply:
https://jira.bigcommerce.com/browse/BCTHEME-61 for sales badge text